### ISSUE TITLE
Enhancement: Preserve quickdial and vanity attributes from old phonebook

### ISSUE THREAD

**Issue by derwok** [andig/carddav2fb#119]
**TL;DR: I'll sent a PR for this enhancement**

The phone book with index "0" allows to specify quickdials and vanity strings for phone numbers
![grafik](https://user-images.githubusercontent.com/4393698/52591699-dbb16280-2e44-11e9-9216-6e75e194fc21.png)

These attributes are proprietary to Fritzbox and not part of CardDav specification. When the CardDav editor allows custom fields (like e.g., OwnCloud/NextCloud), the carddav2fb script expects these attributes to be stored in the CardDav fields `X-FB-QUICKDIAL`and `X-FB-VANITY`.

But when you are forced to a CardDav editor that does not allow to specify own attributes, you can only set quickdial and vanity inside the Fritz!Box Web phonebook editor. In this case the quickdial and vanity strings get lost which each run of carddav2fb script.

Instead of overwriting the phone book blindly, we should rather
* Read in the old phonebook
* Extract specified quickdial and vanity string of contact numbers
* Restore these attributes in the new phone book if CardDavUID and phonenumber are identical
* Then upload the patched new phonebook to the FritzBox

**Comment by BlackSenator** [comment]
Regarding RFC 6350, I do not know why a CardDAV server should not allow user-specific fields. Personally reluctant to me, if the CardDAV does not always remain the leading system.

**Comment by derwok** [comment]
The problem is not the *server* but - as I've written: the CardDav *editor*! Call it "user interface". For example I create/edit most of my contacts via iPhone. Sometimes I use the Contacts software installed on my MacBook - both of them do not support to create/edit these new fields.

Last not least: introducing these proprietary `X-FB-QUICKDIAL`and `X-FB-VANITY` fields does not really make sense to me, as only FritzBox needs them and already has a nice editor to set them. And it even does consistency checks that no duplicates are given.

So the new approach is
* Zero config for users
* Just works out of the box
* Keeps Fritzbox know-how where it belongs: on the FritzBox

My 2cts.  :-)

**Comment by BlackSenator** [comment]
I understand your points, but I'm not sure if the feature is really used that much. There were just two reactions to my survey in the ip-phone forum:
https://www.ip-phone-forum.de/threads/carddav2fb-neue-version-kurzwahl-und-vanity-nummern-erforderlich.301898/
It was my opinion, who really must have it, can also go the way (once) to deposit the data in the CardDAV server. It is nearly "once" because, both attributes are obviously important, belonging to rarely or hardly changing contacts - so no daily changing...
